File: backend/app/models/resume.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, JSON
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from ..database import Base

class Resume(Base):
    __tablename__ = "resumes"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"))
    file_path = Column(String)
    parsed_text = Column(String)
    skills = Column(JSON)
    projects = Column(JSON)
    experience = Column(JSON)
    achievements = Column(JSON)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    user = relationship("User", backref="resumes")
    interviews = relationship("InterviewSession", back_populates="resume")

File: backend/models/Resume.js
const mongoose = require('mongoose');

const resumeSchema = new mongoose.Schema({
  user_id: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
  file_path: String,
  parsed_text: String,
  skills: mongoose.Schema.Types.Mixed,
  projects: mongoose.Schema.Types.Mixed,
  experience: mongoose.Schema.Types.Mixed,
  achievements: mongoose.Schema.Types.Mixed,
}, { timestamps: { createdAt: 'created_at', updatedAt: false } });

module.exports = mongoose.model('Resume', resumeSchema);
